From 48142b119e98e912535475d8b47bea4d9c7811c7 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Mon, 26 Feb 2018 17:54:17 -0500
Subject: [PATCH 193/195] prueth: replace for loop for writing mask with
 memcpy_toio()

Replace the for loop for writing mask value to PRU memory
by memcpy_toio()

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/prueth.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/ti/prueth.c b/drivers/net/ethernet/ti/prueth.c
index c612d36..d050f1e 100644
--- a/drivers/net/ethernet/ti/prueth.c
+++ b/drivers/net/ethernet/ti/prueth.c
@@ -3041,11 +3041,10 @@ static void prueth_sw_set_rx_mode(struct prueth_emac *emac)
 	void __iomem *dram1 = prueth->mem[PRUETH_MEM_DRAM1].va;
 	struct netdev_hw_addr *ha;
 	u8 hash;
-	int i;
 
-	for (i = 0; i < 6; i++)
-		writeb(prueth->sw_mc_mac_mask[i],
-		       dram1 + MULTICAST_FILTER_MASK + i);
+	/* first copy the mask */
+	memcpy_toio(dram1 + MULTICAST_FILTER_MASK,
+		    &prueth->sw_mc_mac_mask[0], ETH_ALEN);
 
 	if (ndev->flags & IFF_ALLMULTI) {
 		writeb(MULTICAST_FILTER_DISABLED,
-- 
1.9.1

